blockchain/state_manager/src/lib.rs

```diff
@@ -1139,8 +1150,10 @@ where
         .fuse();
 
         // Search backwards for message.
-        let mut search_back_poll = task::spawn::<_, Result<_, Error>>(async move {
-            let back_tuple = task.await?;
+        let mut search_back_poll = tokio::task::spawn(async move {
+            let back_tuple = task.await.map_err(|e| {
+                Error::Other(format!("Could not search backwards for message {:?}", e))
```

Comment:
a type implementing `Error` also implements `Display` if I recall correctly
```suggestion
                Error::Other(format!("Could not search backwards for message {e}"))
```
or maybe `to_string` may be needed. No need for a debug display unless I'm missing something.

Same applies to some other occurrences in the PR.

Reply:
Fixed